samples: counter: Add err check on set_alarm call

erwango opened the pull request:
Error checking was missing following call to
counter_set_channel_alarm.
This was reported in coverity report 203523.

Fixes #18374

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

pizi-nordic commented on samples/drivers/counter/alarm/src/main.c:
```diff
@@ -23,6 +23,7 @@ static void test_counter_interrupt_fn(struct device *counter_dev,
 	u64_t now_usec = counter_ticks_to_us(counter_dev, now_ticks);
 	int now_sec = (int)(now_usec / USEC_PER_SEC);
 	struct counter_alarm_cfg *config = user_data;
+	int err = 0;
```
Why err is initialized here?

erwango replied:
Blindly copied code from the main function below, but not needed indeed.

erwango committed:
samples: counter: Add err check on set_alarm call

Error checking was missing following call to
counter_set_channel_alarm.
This was reported in coverity report 203523.

Fixes #18374

Additionally remove unneeded err initialization in main function

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
